src/vm/database/sqlite.rs

```diff
@@ -296,23 +329,96 @@ impl <'a> ContractDatabase <'a> {
         }
     }
 
-    pub fn get_simmed_block_height(&self) -> Result<i128> {
-        let block_height: (i128) =
+    pub fn get_simmed_block_height(&self) -> Result<u64> {
+        let block_height: (i64) =
             self.query_row(
-                "SELECT simmed_block_height FROM simmed_block_table LIMIT 1",
+                "SELECT block_height FROM simmed_block_table ORDER BY block_height DESC LIMIT 1",
                 NO_PARAMS,
                 |row| row.get(0))
             .expect("Failed to fetch simulated block height");
 
-        Ok(block_height)
+        u64::try_from(block_height)
+            .map_err(|_| Error::new(ErrType::Arithmetic("Overflowed fetching block height".to_string())))
     }
 
-    pub fn set_simmed_block_height(&mut self, block_height: i128) {
-        self.execute(
-            "UPDATE simmed_block_table SET simmed_block_height = ?",
-            &[block_height]);
+    pub fn get_simmed_block_time(&self, block_height: u64) -> Result<u64> {
+        let block_height = i64::try_from(block_height).unwrap();
+        let block_time: (i64) = 
+            self.query_row(
+                "SELECT block_time FROM simmed_block_table WHERE block_height = ? LIMIT 1",
+                &[block_height],
+                |row| row.get(0))
+            .expect("Failed to fetch simulated block time");
+
+        u64::try_from(block_time)
+            .map_err(|_| Error::new(ErrType::Arithmetic("Overflowed fetching block time".to_string())))
+    }
+
+    pub fn get_simmed_block_header_hash(&self, block_height: u64) -> Result<Vec<u8>> {
```

Comment:
Sorry, I didn't catch this earlier.  Can this return a `BurnchainHeaderHash` instead of a `Vec<u8>`?  This is in `burnchains::BurnchainHeaderHash`.